2021-APR /answer.md

```diff
@@ -0,0 +1,168 @@
+<!-- @import "[TOC]" {cmd="toc" depthFrom=1 depthTo=6 orderedList=false} -->
+
+# AACS1084 October 2021 Answers
+
+[Link to the paper](https://eprints.tarc.edu.my/17763/1/AACS1084.pdf)
+
+- [Question 1](#question-1)
+- [Question 2](#question-2)
+- [Question 3](#question-3)
+- [Question 4](#question-4)
+
+## Answers
+
+### Question 1
+
+a) (i)
+
+```c
+struct Singer {
+  char name[30];
+  char gender;
+  char nationality[30];
+};
+```
+
+b)
+
+```c
+typedef struct {
+  int id;
+  char title[30];
+  struct Singer singer;
+  double price;
+} Album;
+```
+
+c)
+
+```c
+Album cd[2] = {
+  {1001, "Happy Moments", "Jane Berry", 'F', "American", 67.90},
+  {1002, "Dreams", "Joseph Chan", 'M', "Taiwanese", 59.90}
+};
+```
+
+d)
+
+```c
+strcpy(cd[1].singer.nationality, "Singaporean");
+```
+
+e)
+
+```c
+for (int i = 0; i < 2; i++) {
+  printf("%d\n", cd[i].id);
+  printf("%s\n", cd[i].title);
+  printf("%s\n", cd[i].singer.name);
+  printf("%f\n", cd[i].price);
+}
+system("pause");
```

Comment:
1. Don't have labels, like ID:, Title: etc., follow the output as in the question
2. Price use `RM%.2f`
3. use `\n\n` for the last printf to separate the details of each album